### ISSUE TITLE
Recommended aproach for building external conan recipes in CI

### ISSUE THREAD

**Issue by annulen** [conan-io/conan-package-tools#480]
I'd like to build a package from Conan Center in CI with custom choice of CONAN_OPTIONS and compilers. Is it there any easy way to make CPT build and test specifically that external package instead of "local" conanfile.py? Or is there any other recommended approach for this case?

**Comment by uilianries** [comment]
CPT is used to generate a matrix of builds. If you are looking for a set of builds, I strongly recommend using lockfiles instead: https://docs.conan.io/en/latest/versioning/lockfiles.html#how-to-use-lockfiles-in-ci

**Comment by annulen** [comment]
No, i think I need to generate build matrix. Otherwise I don't understand how can I obtain such lock files in the first place

**Comment by annulen** [comment]
To clarify : this is not about local builds, but about generating matrix of binary packages in ci for uploading to repository

**Comment by annulen** [comment]
I'm using CPT to generate matrix of builds, for example in my fork of ICU recipe I use [this matrix](https://github.com/qtwebkit/conan-icu/blob/stable/65.1/appveyor.yml)  for ICU, and [this one](https://github.com/qtwebkit/conan-libxml2/blob/stable/2.9.10/appveyor.yml) for libxml2. Now both packages are in Conan Center and it's not possible to synchronize with upstream by rebasing my branch, so I wonder what migration path should be chosen.

**Comment by uilianries** [comment]
In this case no. CPT is projected for local builds. You will need a script to download all files, build and upload.
